datafusion/optimizer/src/optimizer.rs

```diff
@@ -384,11 +380,9 @@ impl Optimizer {
 
                 let result = match rule.apply_order() {
                     // optimizer handles recursion
-                    Some(apply_order) => new_plan.rewrite(&mut Rewriter::new(
```

Comment:
Such a simple change :)

Reply:
I think we owe a significant debt to @peter-toth  for his work on the tree node API to sort out how to handle subqueries